packages/netlify-cms-core/src/actions/editorialWorkflow.ts

```diff
@@ -506,40 +517,47 @@ export function deleteUnpublishedEntry(collection: string, slug: string) {
   };
 }
 
-export function publishUnpublishedEntry(collection: string, slug: string) {
-  return (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
+export function publishUnpublishedEntry(collectionName: string, slug: string) {
+  return async (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
     const state = getState();
     const collections = state.collections;
     const backend = currentBackend(state.config);
     const transactionID = uuid();
-    const entry = selectUnpublishedEntry(state, collection, slug);
-    dispatch(unpublishedEntryPublishRequest(collection, slug, transactionID));
-    return backend
-      .publishUnpublishedEntry(entry)
-      .then(() => {
-        // re-load media after entry was published
-        dispatch(loadMedia());
-        dispatch(
-          notifSend({
-            message: { key: 'ui.toast.entryPublished' },
-            kind: 'success',
-            dismissAfter: 4000,
-          }),
-        );
-
-        dispatch(unpublishedEntryPublished(collection, slug, transactionID));
-        return dispatch(loadEntry(collections.get(collection), slug));
-      })
-      .catch((error: Error) => {
-        dispatch(
-          notifSend({
-            message: { key: 'ui.toast.onFailToPublishEntry', details: error },
-            kind: 'danger',
-            dismissAfter: 8000,
-          }),
-        );
-        dispatch(unpublishedEntryPublishError(collection, slug, transactionID));
-      });
+    const entry = selectUnpublishedEntry(state, collectionName, slug);
+    dispatch(unpublishedEntryPublishRequest(collectionName, slug, transactionID));
+    try {
+      await backend.publishUnpublishedEntry(entry);
+      // re-load media after entry was published
+      dispatch(loadMedia());
+      dispatch(
+        notifSend({
+          message: { key: 'ui.toast.entryPublished' },
+          kind: 'success',
+          dismissAfter: 4000,
+        }),
+      );
+      dispatch(unpublishedEntryPublished(collectionName, slug, transactionID));
+      const collection = collections.get(collectionName);
+      if (collection.has('nested')) {
+        dispatch(loadEntries(collection));
```

Comment:
We need to load all entries since some of them might have been moved after publishing.
Also, if we're in the editor and the current slug changed we need to navigate to the new slug